### ISSUE TITLE
Algorithms defined on CXXGraph::Graph should be broken-out to their own files

### ISSUE THREAD

**Issue by nrkramer** [ZigRazor/CXXGraph#303]
Rationale: `Graph.hpp` is getting cluttered and bloated. By moving the algorithm implementations to their own `.hpp` files, we can reduce unnecessary scrolling and searching through `Graph.hpp`.

**Comment by ZigRazor** [comment]
@nrkramer I agree, but it is also true that i have in mind to create a script to give the possibility to include a single-header file, but this is another issue.
